### ISSUE TITLE
Accept a function for a cy.route response

### ISSUE THREAD

**Issue by paulfalgout** [cypress-io/cypress#152]
Passing response data to a `cy.route` should accept a function, and optionally await promises returned from the function
Doing so enables easier access to properties such as aliased fixtures.

Now:
```js
cy.server()
  .fixture('foos').as('fxFoos')
  .then(function() {
    cy.route('GET', /foo\/\d/, {
      data: {
        foo: _.first(this.fxFoos)
      }
    });
  });
```

```js
cy.server()
   .fixture('foos').as('fxFoos')
   .route('GET', /foo\/\d/, function() {
     return {
       data: {
         foo: _.first(this.fxFoos)
        }
       };
     });
   });
```

Particularly useful for:
```js
function fooRoute(resultCount) {
  _.initial(this.fxFoos, resultCount);
}

cy.server()
   .fixture('foos').as('fxFoos')
   .route('GET', /foo\/\d/, _.partial(fooRoute, 4));
```

**Comment by brian-mann** [comment]
This is done and will likely release this weekend.

You'll be able to set response as a `function` which returns a Promise (which will be awaited) or a value.

Additionally we've updated the `cy.route` signature to take also just take a function which can also return a promise or an object with the typical routing signature.

**Comment by paulfalgout** [comment]
Fantastic

**Comment by brian-mann** [comment]
Fixed in [`0.16.2`](https://github.com/cypress-io/cypress/wiki/Changelog#0162-06112016).
